common/src/main/java/org/apache/iceberg/common/DynConstructors.java

```diff
@@ -82,14 +76,6 @@ public <R> R invoke(Object target, Object... args) {
       return (R) newInstance(args);
     }
 
-    @Override
-    @SuppressWarnings("unchecked")
```

Comment:
this method wasn't marked as deprecated?

Reply:
or did we forget to mark this method as deprecated?

Reply:
I don't see any usage of this method. Since abstract method has been removed. But we still have the [invokeChecked](https://github.com/apache/iceberg/pull/10818/files#diff-bef8d1afda79d9ab9186f9fe726389f6dc25b17b1edc95756ab91f4131216e33R55) method in use at super class. 
Do you think we should safely remove this or mark it as deprecated for next release ? 
WDYT?

Reply:
@findepi can you take a look at this one please?

Reply:
@nastra thanks for the ping. this was overlook. more about this -- https://github.com/apache/iceberg/pull/10818/files#r1706672859